lib/cartopy/tests/test_coding_standards.py

```diff
@@ -133,52 +76,33 @@ def test_license_headers(self):
                             'lib/cartopy/_version.py',
                             'versioneer.py',
                             )
+
         try:
-            last_change_by_fname = self.last_change_by_fname()
+            tracked_files = self.list_tracked_files()
         except ValueError as e:
             # Caught the case where this is not a git repo.
             return pytest.skip('cartopy installation did not look like a git '
                                'repo: ' + str(e))
 
-        failed = False
-        for fname, last_change in sorted(last_change_by_fname.items()):
+        failed = []
+        for fname in sorted(tracked_files):
             full_fname = os.path.join(REPO_DIR, fname)
             root, ext = os.path.splitext(full_fname)
             if ext in ('.py', '.pyx', '.c', '.cpp', '.h') and \
                     os.path.isfile(full_fname) and \
                     not any(fnmatch(fname, pat) for pat in exclude_patterns):
 
-                is_empty = os.path.getsize(full_fname) == 0
+                if os.path.getsize(full_fname) == 0:
+                    # Allow completely empty files (e.g. ``__init__.py``)
+                    continue
 
                 with io.open(full_fname, encoding='utf-8') as fh:
                     content = fh.read()
 
-                is_yearless_license = bool(LICENSE_RE_v2.match(content))
-                years = TestLicenseHeaders.years_of_license_in_file(
-                    content, full_fname)
+                if not bool(LICENSE_RE.match(content)):
+                    failed.append(full_fname)
 
-                if is_empty:
-                    # Allow completely empty files (e.g. ``__init__.py``)
-                    pass
-                elif is_yearless_license:
-                    # Allow new style license (v2).
-                    pass
-
-                # What is left is the old-style (pre 2019) header.
-                elif years is None:
-                    print('The file {} has no valid header license and '
-                          'has not been excluded from the license header '
-                          'test.'.format(fname))
-                    failed = True
-                elif last_change.year > years[1]:
-                    print('The file header at {} is out of date. The last'
-                          ' commit was in {}, but the copyright states it'
-                          ' was {}.'.format(fname, last_change.year,
-                                            years[1]))
-                    failed = True
-
-        if failed:
-            raise ValueError('There were license header failures. See stdout.')
+        assert failed == [], 'There were license header failures.'
```

Comment:
I think it would help to print the names of the files that failed still. Is that somewhere else, or could you add it in here with just a `.format(failed)` or similar appended to the string.

Reply:
Pytest will print it out as part of showing that the list is not empty.

Reply:
Ahh, yes! I forgot it would print the comparison variables. Thanks for clarifying.